src/braket/default_simulator/simulator.py

```diff
@@ -407,7 +407,37 @@ def _formatted_measurements(
         ]
         #  Gets the subset of measurements from the full measurements
         if measured_qubits is not None and measured_qubits != []:
-            measurements = np.array(measurements)[:, measured_qubits].tolist()
+            if all(qubit in range(simulation.qubit_count) for qubit in measured_qubits):
+                measurements = np.array(measurements)[:, measured_qubits].tolist()
+            elif any(qubit in range(simulation.qubit_count) for qubit in measured_qubits):
+                measured_qubits_in_circuit = [
+                    qubit for qubit in measured_qubits if qubit in range(simulation.qubit_count)
+                ]
+                measured_qubits_not_in_circuit = [
+                    qubit for qubit in measured_qubits if qubit not in range(simulation.qubit_count)
+                ]
+
+                # get the list of measurements of qubits with gates on the circuit
+                measurements_in_circuit = np.array(measurements)[
+                    :, measured_qubits_in_circuit
+                ].tolist()
+                # set any measurements of qubits without gates to 0
+                measurements = np.zeros(
+                    (simulation.shots, len(measured_qubits_not_in_circuit)), dtype=int
+                ).tolist()
+
+                # Add measurements of qubits with gates on the circuit
+                for idx in measured_qubits_in_circuit:
+                    measurements[idx] = measurements_in_circuit[idx] + measurements[idx]
+
+                # Add measurements of 0s for qubits with no gates on the circuit
+                for idx, sublist in enumerate(measurements):
+                    if idx not in measured_qubits_in_circuit:
+                        measurements[idx] = measurements_in_circuit[idx] + measurements[idx]
+            else:
+                measurements = np.zeros(
+                    (simulation.shots, len(measured_qubits)), dtype=int
+                ).tolist()
```

Comment:
Do these need to be separate cases? From what I can tell, the second branch covers everything

Reply:
Yeah, you're right. The first and last branch are not needed. In fact, there shouldn't need to be a check if the qubits are in the simulation. I'll simplify this down